include/aws/http/request_response.h

```diff
@@ -293,11 +301,26 @@ AWS_EXTERN_C_BEGIN
 AWS_HTTP_API
 bool aws_http_header_name_eq(struct aws_byte_cursor name_a, struct aws_byte_cursor name_b);
 
+/**
+ * Create a new headers object.
+ * The caller has a hold on the object and must call aws_http_headers_release() when they are done with it.
+ */
 AWS_HTTP_API
 struct aws_http_headers *aws_http_headers_new(struct aws_allocator *allocator);
 
+/**
+ * Acquire a hold on the object, preventing it from being deleted until
+ * aws_http_headers_release() is called by all those with a hold on it.
+ */
 AWS_HTTP_API
-void aws_http_headers_destroy(struct aws_http_headers *headers);
+void aws_http_headers_acquire_hold(struct aws_http_headers *headers);
```

Comment:
Why not just acquire?

Reply:
Yeah, Mike? WHY NOT JUST ACQUIRE.

Reply:
aws_channel has `acquire_hold()` and `release_hold()`, but I just searched and see that vanilla `acquire()` and `release()` are more prevalent